File: src/lua/crc32.lua
local uint8 = 0xff
local uint32 = 0xffffffff
local crc_table = {}

function hex(str)
  return ("%x"):format(str)
end

function xor(x, y) 
  return (~x & y) | (x & ~y) 
end

function even(c)
  if c&1 == 1 then
    c = xor(c, 0x1db710640)
  end
  return c
end

for i=1,256 do
  local c = i - 1
  for j=1,8 do
    c = even(c) >> 1
  end
  crc_table[i] = c
end

function crc32(s)
  local crc = uint32
  for c in s:gmatch('.') do
    crc = xor((crc >> 8), crc_table[xor(crc & uint8, string.byte(c)) + 1])
  end
  return xor(crc, uint32)
end

if #arg == 0 then
  print(string.format("Usage: %s [STRING]", arg[0]))
  os.exit(1)
end
print(hex(crc32(arg[1])))

File: src/python/crc32.py
import sys

def create_table():
    a = []
    for i in range(256):
        k = i
        for j in range(8):
            if k & 1:
                k ^= 0x1db710640
            k >>= 1
        a.append(k)
    return a
 
def crc_update(buf, crc):
    crc ^= 0xffffffff
    for k in buf:
        crc = (crc >> 8) ^ crc_table[(crc & 0xff) ^ k]
    return crc ^ 0xffffffff

if __name__ == "__main__":
    if len(sys.argv) == 1:
        print('Empty input!\n\tUsage: {} [STRING]'.format(sys.argv[0]))
        exit(1)
    crc_table = create_table()
    print(hex(crc_update(sys.argv[1].encode(), 0))[2:])
